icepyx/core/spatial.py

```diff
@@ -272,33 +324,29 @@ def validate_polygon_list(spatial_extent):
                 spatial_extent, len(spatial_extent), spatial_extent[1]
             )
 
-    extent_type = "polygon"
     polygon = [float(i) for i in spatial_extent]
 
-    gdf = geodataframe(extent_type, polygon, file=False)
-    spatial_extent = gdf.iloc[0].geometry
-
-    return "polygon", spatial_extent, None
+    return "polygon", polygon, None
 
 
 def validate_polygon_file(spatial_extent):
     """
-       Validates the spatial_extent parameter as a polygon from a file.
+    Validates the spatial_extent parameter as a polygon from a file.
 
-       If the spatial_extent parameter contains a valid polygon,
-       returns a tuple containing the Spatial object parameters for the polygon;
+    If the spatial_extent parameter contains a valid polygon,
+    returns a tuple containing the Spatial object parameters for the polygon;
 
-       otherwise, throw an error containing the reason the polygon/polygon file is invalid.
+    otherwise, throw an error containing the reason the polygon/polygon file is invalid.
 
-       Parameters
-       ----------
-       spatial_extent: string
-                        A string representing a geospatial polygon file (kml, shp, gpkg)
-                        * must provide full file path
-                        * recommended for file to only contain 1 polygon.
-                            * if multiple polygons, only the first polygon is selected at this time.
+    Parameters
+    ----------
+    spatial_extent: string
+                     A string representing a geospatial polygon file (kml, shp, gpkg)
```

Comment:
At L361, it says `if spatial_extent.split(".")[-1] in ["kml", "shp", "gpkg"]:` which restricts the possible file types to KMLs, shapefiles and geopackages. However at L114, the code uses `gpd.read_file(spatial_extent)`, which should mean that any other OGR supported vector file format should work (e.g. GeoJSON, FlatGeoBuf, etc).

Do you think it's possible to remove the hardcoded file extension check at L361? Maybe a try-except statement to see if the file can be read directly into a `geopandas.GeoDataFrame`?

Reply:
Great points!

The file type list limitations is a combination of what [early] NSIDC tutorials explicitly stated were submissable and formats that I (as of 3+ years ago) was familiar with or confident testing. Theoretically, any "vector-based spatial data file" (as stated in [this notebook](https://github.com/nsidc/NSIDC-Data-Access-Notebook/blob/master/notebooks/Customize%20and%20Access%20NSIDC%20Data.ipynb) can be submitted to the ICESat-2 subsetter, which should jibe with:

> any other OGR supported vector file format should work (e.g. GeoJSON, FlatGeoBuf, etc)

I'm hesitant to remove the file-type restrictions until we have tried submitting (and ideally added to the test suite) at least a few of the other more common vector file formats. How would you feel about converting this to an issue for future work?

Reply:
Yes, let's track the OGR vector file support in a separate issue. My impression is that we're not submitting the vector file (`.shp`, `.geojson`, etc) directly, but the WKT geometry parsed from the vector file, so any geopandas readable file should work in theory.

Reply:
See #384. During ordering, the file itself (rather than the WKT geometry) is passed.